src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -583,6 +584,20 @@ ReactDOMComponent.Mixin = {
         );
         didWarnShadyDOM = true;
       }
+
+      if (__DEV__) {
+        if (this._namespaceURI === DOMNamespaces.html) {
+          warning(
+            !(ownerDocument.createElement(this._tag) instanceof window.HTMLUnknownElement) ||
+              isCustomComponent(this._tag, props) || voidElementTags[this._tag],
```

Comment:
Intuitively you either have a whitelist of all the currently known elements and the warning is then used to raise awareness that you have made a typo. Or you don't have a whitelist and the warning is used to indicate that you're rendering an element that isn't understood by the current browser.

There are many tags not supported by every browser, whitelisting just `menuitem` (and self-closing tags) seems like an arbitrary in-between. IMHO you should pick one or the other (@aweary?).